backend/db/utils.py

```diff
@@ -462,15 +462,59 @@ def database_exists(con: Connection, db_name: str) -> bool:
     return len(result) == 1
 
 
-def run_sql(con: Connection, query: str, params: Dict = {}) -> CursorResult:
+def run_sql(con: Connection, query: str, params: Dict[str, Any] = {}) -> CursorResult:
     """Run a sql command"""
     query = text(query) if not isinstance(query, TextClause) else query
     return con.execute(query, params) if params else con.execute(query)
 
 
+def sql_handle_none_to_null(query: str, params: Dict[str, Any], table: str, schema=SCHEMA) -> str:
+    """Convert None to NULL for certain fields.
+
+    SqlAlchemy w/ Postegres is finicky when accepting None as a NULL value for certain field types, e.g. numeric ones.
+    Will find any numeric fields on table and convert None to NULL.
+    """
+    # Find numeric fields
+    field_data_types: Dict[str, str] = get_field_data_types(table, schema)
+    numeric_fields: Set[str] = set([k for k, v in field_data_types.items() if v in PG_DATATYPES_BY_GROUP['numeric']])
+    # Figure out which fields need replacement
+    fields_with_none: List[str] = [k for k, v in params.items() if v is None]
+    # todo: this is really inefficient: O(n^2)
+    numeric_fields_with_none: List[str] = []
+    for fld1 in fields_with_none:
+        for fld2 in numeric_fields:
+            if fld1.startswith(fld2):
+                numeric_fields_with_none.append(fld2)
+                # numeric_fields.remove(fld2)  # would be performance optimization; but causes set size change err
+    # Make replacements
+    query2 = query
+    for fld in numeric_fields_with_none:
+        query2 = query2.replace(f'"{fld}" = v.{fld}',
+            f'"{fld}" = CASE WHEN v.{fld}::text = \'None\' THEN NULL ELSE v.{fld}::double precision END')
+    return query2
+
+
+# todo: schema: Ideally, I should make sure that any usages of this function are passing `schema`.
+def run_sql_update(
```

Comment:
## `run_sql_update()`
If you get an error related to NULL while doing an update query, use this in combination with the `handle_none_as_null_on_table` param.
Basically, I want to ensure that if the user deletes or submits empty text / null for a metadata field, we can support that. Normally it is not an issue, except for numeric fields. 

This was a pain in the ass to fix. I tried different things, like `sqlalchemy.null()` and 
`sqlalchemy.Float().bind_processor(dialect=con.dialect)(None)`, but fixed another way. Now I question whether there's ever a situation we really _need_ this support. I'm looked at the numeric fields, and I'm not sure if they can ever go from a non-null to a null value. Oh well.